src/main/java/org/opensearch/ad/transport/AnomalyResultTransportAction.java

```diff
@@ -304,6 +303,9 @@ class PageListener implements ActionListener<CompositeRetriever.Page> {
 
         @Override
         public void onResponse(CompositeRetriever.Page entityFeatures) {
+            if (pageIterator.hasNext()) {
```

Comment:
If there are too many pages and not reach the last page until current job interval ends, will current job still iterate next page while next job triggers?

Reply:
yes